java-generator/core/src/test/java/io/fabric8/java/generator/ApprovalTest.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import java.util.ArrayList;
+import java.util.List;
+import org.approvaltests.Approvals;
+import org.junit.jupiter.api.Test;
+
+class ApprovalTest {
+
+    KubernetesClient client = new DefaultKubernetesClient();
```

Comment:
It would be advisable to use the KubernetesMockServer Extension.
With this approach, the client might target a real cluster. In addition, it's never closed.

If not, maybe tear it down, and make sure that the client is initialized with an empty configuration. The current tests seem harmless, but you never know what might end up here.

Reply:
👍  removed and used the `Serialization` class